rust/foxglove_data_loader/examples/ndjson.rs

```diff
@@ -0,0 +1,374 @@
+//! This example data loader parses newline-separated lines of JSON of 3 forms:
+//!
+//! {"type":"info","start":0,"end":30,"sphere_count":360,"temperature_count":120}
```

Comment:
I would probably move the `info` record to the end of the file. If you imagine that the file is written by appending new temperature readings, it is somewhat unexpected to have the summary record at the start of the file as the end time is not known yet.

Do we actually need that record? You currently read the whole file to memory, so you could also derive start and end time from the first and last line?